bundles/org.openhab.binding.homeconnect/src/main/java/org/openhab/binding/homeconnect/internal/client/HomeConnectEventSourceListener.java

```diff
@@ -0,0 +1,238 @@
+
+        eventSourceMonitorFuture = createMonitor(scheduler);
+    }
+
+    public void onEvent(InboundSseEvent inboundEvent) {
+        @Nullable
+        String id = inboundEvent.getId();
+        @Nullable
+        String type = inboundEvent.getName();
+        @Nullable
+        String data = inboundEvent.readData();
+
+        lastEventReceived = now();
+
+        @Nullable
+        EventType eventType = valueOfType(type);
+        if (eventType != null) {
+            mapEventSourceEventToEvent(haId, eventType, data).forEach(event -> {
+                eventQueue.add(event);
+                logger.debug("Received event ({}): {}", haId, event);
+                try {
+                    eventListener.onEvent(event);
+                } catch (Exception e) {
+                    logger.error("Could not publish event to Listener!", e);
+                }
+            });
+        } else {
+            logger.warn("Received unknown event source type! haId={}, id={}, type={}, data={}", haId, id, type, data);
+        }
+    }
+
+    public void onComplete() {
+        logger.debug("Event source listener channel closed ({}).", haId);
+
+        client.unregisterEventListener(eventListener, true);
+
+        try {
+            eventListener.onClosed();
+        } catch (Exception e) {
+            logger.error("Could not publish closed event to listener ({})!", haId, e);
+        }
+        stopMonitor();
+    }
+
+    public void onError(Throwable error) {
+        @Nullable
+        String throwableMessage = error.getMessage();
+        String throwableClass = error.getClass().getName();
+
+        logger.debug("Event source listener connection failure occurred. haId={}, throwable={}, throwableMessage={}",
+                haId, throwableClass, throwableMessage);
+
+        client.unregisterEventListener(eventListener);
+
+        try {
+            if (throwableMessage != null
+                    && throwableMessage.contains(String.valueOf(HttpStatus.TOO_MANY_REQUESTS_429))) {
+                logger.warn(
+                        "More than 10 active event monitoring channels was reached. Further event monitoring requests are blocked. haId={}",
+                        haId);
+                eventListener.onRateLimitReached();
+            } else {
+                // The SSE connection is closed by the server every 24 hours.
+                // When you try to reconnect, it often fails with a NotAuthorizedException (401) for the next few
+                // seconds. So we wait few seconds before trying again.
+                if (error instanceof NotAuthorizedException) {
+                    logger.debug(
+                            "Event source listener connection failure due to unauthorized exception : wait 5 seconds... haId={}",
+                            haId);
+                    try {
+                        Thread.sleep(5000);
+                    } catch (InterruptedException e1) {
+                    }
```

Comment:
No this is not handled by a framework thread.
I believe it is handled by a thready created by JAX-RS SSE.
Scheduling a task with a delay here will add complexity to the code IMHO.

Reply:
I would consider JAX-RS SSE as part of the framework. I couldn't figure out at first glance which thread pool is used or if any is used at all. If it is a shared thread pool, `sleep` must not be used. You could break into it and check the stack trace.

I don't think it would make the code much more complicated.

Reply:
Would it be ok to change it to:

```
                if (error instanceof NotAuthorizedException) {
                    logger.debug(
                            "Event source listener connection failure due to unauthorized exception : wait 5 seconds... haId={}",
                            haId);

                    scheduler.schedule(() -> eventListener.onClosed(), 5, TimeUnit.SECONDS);
                } else {
                    eventListener.onClosed();
                }
        
```

Reply:
Sure!